R/zchunk_L224.heat.R

```diff
@@ -65,185 +73,341 @@ module_energy_L224.heat_DISABLED <- function(command, ...) {
+        filter(region %in% heat_region$region) -> L224.SubsectorShrwt_heat
+    }
+
+    if(any(!is.na(A24.subsector_shrwt$year.fillout))){
+      A24.subsector_shrwt %>%
+        filter(!is.na(year.fillout)) %>%
+        write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]], GCAM_region_names = GCAM_region_names ) %>%
+        filter(region %in% heat_region$region) -> L224.SubsectorShrwtFllt_heat
+    }
+
+    # Subsector shareweight interpolation of district heat sectors
+    if(any(is.na(A24.subsector_interp$to.value))){
+      A24.subsector_interp %>%
+        filter(is.na(to.value)) %>%
+        write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorInterp"]], GCAM_region_names = GCAM_region_names) %>%
+        filter(region %in% heat_region$region) -> L224.SubsectorInterp_heat
+    }
+
+    if(any(!is.na(A24.subsector_interp$to.value))){
+      A24.subsector_interp %>%
+        filter(!is.na(to.value)) %>%
+        write_to_all_regions(LEVEL2_DATA_NAMES[["SubsectorInterpTo"]], GCAM_region_names = GCAM_region_names) %>%
+        filter(region %in% heat_region$region) -> L224.SubsectorInterpTo_heat
+    }
+
+    # Identification of stub technologies of district heat
+    # Note: assuming that technology list in the shareweight table includes the full set (any others would default to a 0 shareweight)
+    A24.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]], GCAM_region_names = GCAM_region_names) %>%
+      rename(stub.technology = technology) %>%
+      filter(region %in% heat_region$region) -> L224.StubTech_heat
+
+    # Coefficients of global technologies
+    # L224.GlobalTechCoef_heat: Energy inputs and coefficients of global technologies for district heat
+    A24.globaltech_coef %>%
+      gather(year, coef, -supplysector, -subsector, -technology, -minicam.energy.input) %>%
+      mutate(year = as.integer(year)) -> A24.globaltech_coef
+
+    A24.globaltech_coef %>%
+      select(supplysector, subsector, technology, minicam.energy.input) %>%
+      distinct %>%
+      # Interpolate to all years
+      repeat_add_columns(tibble(year = c(HISTORICAL_YEARS, FUTURE_YEARS))) %>%
+      left_join(A24.globaltech_coef, by = c("supplysector", "subsector", "technology", "minicam.energy.input", "year")) %>%
+      mutate(coef = round(approx_fun(year, value = coef, rule = 1), energy.DIGITS_COEFFICIENT)) %>%
+      filter(year %in% MODEL_YEARS) %>%
+      select(sector.name = supplysector, subsector.name = subsector, technology, minicam.energy.input, year, coefficient = coef) -> L224.GlobalTechCoef_heat
+
+    # Costs of global technologies
+    # L224.GlobalTechCost_heat: Costs of global technologies for district heat
+    A24.globaltech_cost %>%
+      gather(year, input.cost, -supplysector, -subsector, -technology, -minicam.non.energy.input) %>%
+      mutate(year = as.integer(year)) -> A24.globaltech_cost
+
+    A24.globaltech_cost %>%
```

Comment:
Could we merge these two pipelines?

Reply:
@kvcalvin the first one puts A24.globaltech_cost into long format and then in the second pipeline it joins on this again (so needs to be long). I could just move the first pipeline to the read data section, which I think has been a convention with 'gather'ing input files? Same story for the ones below.

Reply:
Actually the one* below. Other two can be merged!